python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -29,6 +28,33 @@
 # entry/init points.
 logger = logging.getLogger(__name__)
 
+def _get_current_device_type() -> str:
+    """
+    Check the current device type (GPU or NPU) and return its name.
+
+    Returns:
+        A string indicating the device type, either 'cuda' for GPU or 'npu' for NPU.
+    """
+    import torch
+
+    # Get the current device type
+    if torch.cuda.is_available():
+        return "cuda"
+    elif hasattr(torch, "npu") and torch.npu.is_available():
+        return "npu"
+    else:
+        raise RuntimeError("No supported accelerator device (GPU or NPU) found.")
+
+
+# Determine which communicator to use based on the current device type
+device_type = _get_current_device_type()
+if device_type == "npu":
+    from ray.experimental.channel.nccl_group import _NcclGroup
+
+else:
+    from ray.experimental.channel.hccl_group import _HcclGroup as _NcclGroup
```

Comment:
hmm, this looks like a hack. Do you plan to change to a cleaner approach?

Reply:
OK, I just remove this hack and left a comments in the test. After we refactor the channel we can have better solution.